student.rb

```diff
@@ -9,6 +9,11 @@ def initialize(age, classroom, name = 'Unknown', parent_permission: true)
   def play_hooky
     '¯(ツ)/¯'
   end
+
+  def assign_classroom(classroom)
+    @classroom = classroom
+    classroom.add_student(self) unless classroom.students.include?(self)
+  end
```

Comment:
Just as I suggested earlier since we will also need to add read access for the classroom since we are accessing it here😉.  

`attr_reader :classroom`